code/modules/ai/datum_node.dm

```diff
@@ -0,0 +1,21 @@
+//A datum that stores information about this node, the actual effect nodes have these created and referenced
+
+/datum/ai_node
+	var/obj/effect/ai_node/parentnode //The effect node this is attached to
+	var/list/adjacent_nodes = list() // list of adjacent landmark nodes
+	var/list/weights = list(ENEMY_PRESENCE = 0, DANGER_SCALE = 0) //List of weights for the overall things happening at this node
+
+//If we wanted to see if it's not set to 0
+/datum/ai_node/proc/weight_not_null(name)
+	if(name in weights && !weights[name])
```

Comment:
```suggestion
	if(weights[name] == 0)
```
Doesn't this work?
Also careful with the `in` operator without parentheses.

Reply:
left to right is okay and `if(!weights[name])` is better